src/main/java/edu/cornell/kfs/fp/batch/service/impl/AwsAccountingXmlDocumentAccountingLineServiceImpl.java

```diff
@@ -0,0 +1,346 @@
+        String[] costCenterGroupValueSplit = StringUtils.splitByWholeSeparatorPreserveAllTokens(costCenterGroupLevelValue,"*", -1);
+        AccountingXmlDocumentAccountingLine ret = getEmptyAccountingXmlDocumentAccountingLine();
+        ret.setChartCode(costCenterGroupValueSplit[0]);
+        ret.setAccountNumber(costCenterGroupValueSplit[1]);
+        ret.setSubAccountNumber(costCenterGroupValueSplit[2]);
+        ret.setObjectCode(costCenterGroupValueSplit[3]);
+        ret.setSubObjectCode(costCenterGroupValueSplit[4]);
+        ret.setProjectCode(costCenterGroupValueSplit[5]);
+        ret.setOrgRefId(costCenterGroupValueSplit[6]);
+
+        return ret;
+    }
+
+    private boolean validateAccount(String chartCode, String accountNumber) {
+        if (StringUtils.isBlank(accountNumber)) {
+            LOG.error(configurationService.getPropertyValueAsString(CuFPKeyConstants.ERROR_ACCOUNT_NUMBER_BLANK));
+            return false;
+        }
+
+        if (StringUtils.equalsIgnoreCase(accountNumber, CuFPConstants.AmazonWebServiceBillingConstants.INTERNAL_KFS_ACCOUNT_DESCRIPTION)) {
+            return true;
+        }
+
+        Account account = accountService.getByPrimaryId(chartCode, accountNumber);
+        if (ObjectUtils.isNull(account)) {
+            LOG.error(String.format(configurationService.getPropertyValueAsString(CuFPKeyConstants.ERROR_ACCOUNT_NOT_FOUND), chartCode, accountNumber));
+            return false;
+        }
+        if (account.isClosed()) {
+            LOG.error(String.format(configurationService.getPropertyValueAsString(CuFPKeyConstants.ERROR_ACCOUNT_CLOSED), chartCode, accountNumber));
+            return false;
+        }
+        if (account.isExpired()) {
+            LOG.error(String.format(configurationService.getPropertyValueAsString(CuFPKeyConstants.ERROR_ACCOUNT_EXPIRED), chartCode, accountNumber));
+            return false;
+        }
+        return true;
+    }
+
+    private boolean validateObjectCode(String chartCode, String objectCodeValue) {
+        if (StringUtils.isBlank(objectCodeValue)) {
+            return false;
+        }
+
+        ObjectCode objectCode = objectCodeService.getByPrimaryIdForCurrentYear(chartCode, objectCodeValue);
+        if (ObjectUtils.isNull(objectCode)) {
+            LOG.error(String.format(configurationService.getPropertyValueAsString(CuFPKeyConstants.ERROR_OBJECT_CODE_NOT_FOUND), chartCode, objectCodeValue));
+            return false;
+        }
+        if (!objectCode.isActive()) {
+            LOG.error(String.format(configurationService.getPropertyValueAsString(CuFPKeyConstants.ERROR_OBJECT_CODE_INACTIVE), chartCode, objectCodeValue));
+            return false;
+        }
+        return true;
+    }
+
+    private boolean validateSubAccountNumber(String chartCode, String accountNumber, String subAccountNumber) {
+        if (StringUtils.isBlank(subAccountNumber)) {
+            return false;
```

Comment:
I think an empty sub account is valid, as it can be blank.  This is also try  for sub object, project, and org ref ID